docs/design/sparse-v2.md

```diff
@@ -0,0 +1,225 @@
+# Sparse Patterns v2 redesign
+
+Authors: [Daniel Ploch](mailto:dploch@google.com)
+
+**Summary:** This Document documents a redesign of the sparse patterns list
+stored in the JJ working copy, in order to facilitate several desirable
+improvements. It covers both the migration path and the planned end state.
```

Comment:
Suggestion: "This Document documents a redesign of the sparse pattern command and it's internal storage format in the JJ working copy, in order to facilitate several desirable improvements for large monorepos."

Reply:
> Why mention large monorepos?

This seems to me like a bunch of clientside improvements for a certain large repo, which is the reason why I choose "monorepo". 

> It seems to apply equally to smaller (but still large) repos like the Linux repo (or do you consider that a monorepo?).

Fair enough. Im going to edit my suggestion to just mention "large repos".

Reply:
Yes, the more generic patterns and the remapping are both certainly more useful in very large repos. But, interestingly, #1896 and #2288 were filed by non-Googlers (and non-monorepo-users even, I think).